modules/top-level/files/submodule.nix

```diff
@@ -22,6 +23,10 @@
       path = lib.mkDefault name;
       type = lib.mkDefault (if lib.hasSuffix ".vim" name then "vim" else "lua");
       # No need to use mkDerivedConfig; this option is readOnly.
-      plugin = helpers.writeLua derivationName config.content;
+      plugin =
+        let
+          writeContent = if config.type == "lua" then helpers.writeLua else pkgs.writeText;
+        in
+        writeContent derivationName config.content;
```

Comment:
I think this is unrelated to this PR? (Incidentally, #1889 fixes this too)

Reply:
That's true. I needed it for a test to pass, because it failed to create a .vim file. I already forgot that I changed it. It's better be fixed outside of this PR.